specs/PermissionManagement.md

```diff
@@ -0,0 +1,467 @@
+} COREWEBVIEW2_PERMISSION_KIND;
+```
+```c#
+namespace Microsoft.Web.WebView2.Core
+{
+    enum CoreWebView2PermissionKind
+    {
+        // Other permission kinds not shown.
+
+        // Indicates permission to automatically download multiple files.
+        AutomaticDownloads = 7,
+
+        // Indicates permission to edit files or folders on the device.
+        FileEditing = 8,
+
+        // Indicates permission to play audio and video automatically on sites.
+        Autoplay = 9,
+
+        // Indicates permission to use fonts on the device.
+        LocalFonts = 10,
+    };
+}
+```
+## Extended PermissionRequestedEventArgs: ShouldPersist
+```
+/// This is a continuation of the `ICoreWebView2PermissionRequestedEventArgs`
+/// interface.
+[uuid(08595a19-44f0-41b1-9ae4-5889f5edadcb), object, pointer_default(unique)]
+interface ICoreWebView2PermissionRequestedEventArgs3:
+    ICoreWebView2PermissionRequestedEventArgs2 {
+  /// The permission state set from the `PermissionRequested` event is persisted
+  /// across sessions by default and becomes the new default behavior for future
+  /// `PermissionRequested` events. Browser heurisitics can affect whether the
+  /// event continues to be raised when the state is persisted. Set the
+  /// `ShouldPersist` property to `FALSE` to not persist the state beyond the
+  /// current request, and to continue to receive `PermissionRequested` events
+  /// for this origin and permission kind.
+  [propget] HRESULT ShouldPersist([out, retval] BOOL* shouldPersist);
+
+  /// Sets the `ShouldPersist` property.
+  [propput] HRESULT ShouldPersist([in] BOOL shouldPersist);
+}
+```
+```c#
+namespace Microsoft.Web.WebView2.Core
+{
+    runtimeclass CoreWebView2PermissionRequestedEventArgs
+    {
+        // Other members not shown.
+
+        // The permission state set from the `PermissionRequested` event is persisted
+        // across sessions by default and becomes the new default behavior for future
+        // `PermissionRequested` events. Browser heurisitics can affect whether the
+        // event continues to be raised when the state is persisted. Set the
+        // `ShouldPersist` property to `FALSE` to not persist the state beyond the
+        // current request, and to continue to receive `PermissionRequested` events
+        // for this origin and permission kind.
+        [interface_name("Microsoft.Web.WebView2.Core.ICoreWebView2PermissionRequestedEventArgs3")]
+        {
+            Boolean ShouldPersist { get; set; };
+        }
+    }
+}
+```
+## SetPermission and GetNonDefaultPermissionCollection
+```
+/// This is the ICoreWebView2 interface for the permission management APIs.
+[uuid(5bdfc5dd-C07a-41b7-bcf2-94020975f185), object, pointer_default(unique)]
+interface ICoreWebView2Profile6 : ICoreWebView2Profile5 {
+  /// Sets permission state for the given permission kind and origin
+  /// asynchronously. The change takes effect immediately and persists
```

Comment:
Chrome doesn't have a callback for this function, so initially I had a PermissionChanged event instead. I removed it after some discussion in the design spec on whether developers ever need to know about the change in real time - i.e. it might be sufficient to call GetNonDefaultPermissionList when the user navigates to an app's permission management page or refreshes that page. What do you think?